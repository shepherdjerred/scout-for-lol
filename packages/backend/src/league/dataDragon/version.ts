import { z } from "zod";
import { first } from "remeda";

const versions = z
  .array(z.string())
  .parse(
    await (
      await fetch("https://ddragon.leagueoflegends.com/api/versions.json")
    ).json(),
  );

export const latestVersion = first(versions) as string;

if (latestVersion === undefined) {
  throw new Error("latest version is undefined");
}
