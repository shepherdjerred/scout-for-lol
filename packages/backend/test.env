VERSION=test
GIT_SHA=test
ENVIRONMENT=dev
DISCORD_TOKEN=test
APPLICATION_ID=test
RIOT_API_TOKEN=test
DATABASE_URL=file:./db.sqlite

# S3 Storage Configuration (optional - leave commented for testing)
# S3_BUCKET=scout-lol-matches-test
# AWS_REGION=us-east-1
# AWS_ACCESS_KEY_ID=your-access-key
# AWS_SECRET_ACCESS_KEY=your-secret-key
# S3_ENDPOINT=https://s3.amazonaws.com
# S3_FORCE_PATH_STYLE=false
