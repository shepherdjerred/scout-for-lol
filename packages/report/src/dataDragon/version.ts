import { z } from "zod";
import { first } from "remeda";

const versions = z
  .array(z.string())
  .parse(await (await fetch("https://ddragon.leagueoflegends.com/api/versions.json")).json());

const firstVersion = first(versions);

if (firstVersion === undefined) {
  throw new Error("latest version is undefined");
}

export const latestVersion = firstVersion;
