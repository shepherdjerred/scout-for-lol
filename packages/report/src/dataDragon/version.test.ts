import { latestVersion } from "./version.js";
import { test, expect } from "bun:test";

test("should be able to get version", () => {
  expect(latestVersion).toMatchSnapshot();
});
