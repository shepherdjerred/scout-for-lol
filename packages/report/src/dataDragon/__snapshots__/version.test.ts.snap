export const snapshot = {};

snapshot[`should be able to get version 1`] = `"15.1.1"`;
